### ISSUE TITLE
feat: use uglify-js for es5 bundles

### ISSUE THREAD

**Issue by L2jLiga** [angular/angular-cli#14651]
# 🚀 Feature request
I think it would be nice to use `uglify-js` for es5 bundles as it more beneficial.

### Command (mark with an `x`)

```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Description
Terser is great minificator which supports es2015, but on "es5 battlefields" uglify-js is winner 😄 
It's faster and may produce same or better result.

### Describe the solution you'd like
CLI uses uglify-js for es5 bundles

### Describe alternatives you've considered
🤔

**Comment by destus90** [comment]
> but on "es5 battlefields" uglify-js is winner
how have you understand it? Terser is a fork of uglify-js and it shoud use the same algorithms. Am I wrong?

**Comment by L2jLiga** [comment]
> how have you understand it?

I've just observe on progress of both packages. Goal of Terser is to be "A JavaScript parser and mangler/compressor toolkit for ES6+."
When uglify-js goal is to be "parser, minifier, compressor and beautifier toolkit" for ES5


> Terser is a fork of uglify-js and it shoud use the same algorithms.

Terser was forked long time ago and for now they are have different ways of moving forward. For now we have 2 projects with a lot of different commits.

**Comment by alan-agius4** [comment]
@L2jLiga, thanks for opening this. 

We did use `uglify-js` up till version 7 (approx a year ago), when we switched over to `terser`, the main reasons for this were change was that terser was faster at times up to 60% on ES5 bundles and also solved some bugs which were being encountered with `uglify-js`.

At this point in time, I don't see any reason why we should consider changing the JS minifier again. For several reasons.

1) Replacing uglify with terser did solve a lot of issues, both from a performance stand point and from code minification issues.
2) Adding another minifier as a dependency would also increase the tarball size which is not desired. 
3) Might cause regressions which as there is no guarantee that all the bugs that were present earlier which terser solved have been indeed solved by `uglify-js`.

At the point, I suggest to close this, and re-open it maybe for the version 9 timeframe if we see that this still applies at that point.
